Automattic/jetpack
# Blocks: add Jetpack block elements to WPML configuration

#### Changes proposed in this Pull Request:

This is a replay of #21077

> To demonstrate how to add Gutenberg Blocks to the Jetpack WPML file, I've added a selection of blocks and set them as Translateable via WPML's Advanced Translation Editor. 
> 
> These blocks are related to form fields and include:
> 
> ```
> "jetpack/field-name"
> "jetpack/field-email"
> "jetpack/field-telephone"
> "jetpack/field-textarea"
> "jetpack/field-radio"
> "jetpack/field-select"
> "jetpack/button"
> "jetpack/field-consent"
> ```
> 
> Full instructions for registering Gutenberg Blocks in the WPML config file can be found [here](https://wpml.org/documentation/support/language-configuration-files/#gutenberg-blocks)

*************

**For future reference (not applicable in this case):**

Beyond strings, there might be:

* Some references to other posts/terms => In the translation, we should see the translation of the referenced post/term
* Some internal links => In the translation, we should see the links converted to point to the translated things.
* Images => Install WPML Media and, after "translating" the images, check that it's replaced in the translated post

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

Testing this requires an active WPML subscription.

If you have one, you can try the following:

1. After configuring WPML, make sure 'WPML's Advanced Translation Editor' is selected under 'WPML' -> 'Settings'
2. Create new page/post and include all relevant block fields in the original language page/post
3. Where block field is customizable by user, make a small change (Default content wrapped in 'gettext' is translatable via String Translation)
4. Translate page/post via WPML's Advanced Translation Editor (click the '+' icon next to your 'Page' or 'Post' to open and make translations in the editor). Form fields included in XML file should be available to translate 
5. Once translations are made, check if everything is properly translated on the frontend
